packages/vite/src/node/plugins/resolve.ts

```diff
@@ -152,6 +153,28 @@ export function resolvePlugin(resolveOptions: InternalResolveOptions): Plugin {
         scan: resolveOpts?.scan ?? resolveOptions.scan,
       }
 
+      const resolveSubpathImports = (id: string, importer?: string) => {
+        if (!importer || !id.startsWith(subpathImportsPrefix)) return
+        const basedir = path.dirname(importer)
+        const importerPkgJson = lookupFile(basedir, ['package.json'], {
```

Comment:
Should this use `loadPackageData` instead? It looks like `loadPackageData` has an internal cache, so may be faster